app/controllers/assessments_controller.rb

```diff
@@ -823,4 +812,114 @@ def valid_asmt_tar(tar_extract)
     end
     [asmt_rb_exists && asmt_yml_exists && (!asmt_name.nil?), asmt_name]
   end
+
+  # methods for sending different file packages depending on what button was clicked
+def exportEverything
+  base_path = Rails.root.join("courses", @course.name).to_s
+  asmt_dir = @assessment.name
+  begin
+      # Pack assessment directory into a tarball.
+      tarStream = StringIO.new("")
+      Gem::Package::TarWriter.new(tarStream) do |tar|
+        tar.mkdir asmt_dir, File.stat(File.join(base_path, asmt_dir)).mode
+        Dir[File.join(base_path, asmt_dir, "**", "*")].each do |file|
+          mode = File.stat(file).mode
+          relative_path = file.sub((/^#{Regexp.escape base_path}\/?/), "")
+
+          if File.directory?(file)
+            tar.mkdir relative_path, mode
+          else 
+            tar.add_file relative_path, mode do |tarFile|
+              File.open(file, "rb") { |f| tarFile.write f.read }
+            end
+          end
+        end
+      end
+      tarStream.rewind
+      tarStream.close
+      send_data tarStream.string.force_encoding("binary"), filename: "#{@assessment.name}_#{Time.now.strftime('%Y%m%d')}.tar", content_type: "application/x-tar"
+    rescue SystemCallError => e
+      flash[:error] = "Unable to generate tarball -- #{e.message}"
+      redirect_to(:action => 'exportOptions') && return
+    rescue StandardError => e
+      flash[:error] = "Unable to generate tarball -- #{e.message}"
+      redirect_to(:action => 'exportOptions') && return 
+    else
+
```

Comment:
Empty else clause? Did you forget the success message?

Reply:
the "success message" is the download itself so ill delete the clause